token-minter/sources/modules/collection_components.move

```diff
@@ -34,277 +36,180 @@ module minter::collection_components {
     /// Collection properties does not exist on this object.
     const ECOLLECTION_PROPERTIES_DOES_NOT_EXIST: u64 = 1;
 
-    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
-    struct CollectionProperties has copy, drop, key {
-        /// Determines if the creator can mutate the collection_properties's description
-        mutable_description: bool,
-        /// Determines if the creator can mutate the collection_properties's uri
-        mutable_uri: bool,
-        /// Determines if the creator can mutate token descriptions
-        mutable_token_description: bool,
-        /// Determines if the creator can mutate token names
-        mutable_token_name: bool,
-        /// Determines if the creator can mutate token properties
-        mutable_token_properties: bool,
-        /// Determines if the creator can mutate token uris
-        mutable_token_uri: bool,
-        /// Determines if the creator can change royalties
-        mutable_royalty: bool,
-        /// Determines if the creator can burn tokens
-        tokens_burnable_by_creator: bool,
-        /// Determines if the creator can transfer tokens
-        tokens_transferable_by_creator: bool,
-    }
-
-    #[event]
-    /// Event emitted when CollectionRefs are created.
-    struct CreateCollectionRefs has drop, store {
-        mutator_ref_exists: bool,
-        royalty_mutator_ref_exists: bool,
-        extend_ref_exists: bool,
-    }
-
-    #[event]
-    /// Event emitted when CollectionProperties are created.
-    struct InitCollectionProperties has drop, store {
-        mutable_description: bool,
-        mutable_uri: bool,
-        mutable_token_description: bool,
-        mutable_token_name: bool,
-        mutable_token_properties: bool,
-        mutable_token_uri: bool,
-        mutable_royalty: bool,
-        tokens_burnable_by_creator: bool,
-        tokens_transferable_by_creator: bool,
-    }
-
-    public fun create_refs(
-        constructor_ref: &ConstructorRef,
-        mutable_description: bool,
-        mutable_uri: bool,
-        mutable_royalty: bool,
-    ): Object<CollectionRefs> {
-        let collection_signer = object::generate_signer(constructor_ref);
+    /// This function creates all the refs to extend the collection, mutate the collection and royalties.
+    public fun create_refs_and_properties(constructor_ref: &ConstructorRef): Object<CollectionRefs> {
```

Comment:
should we move the initial default property boolean value to be passed in here instead? btw the boolean param for `create_default_properties(true)` isn't needed cuz we always default them to true in the code. it might make sense for the testing thou to accept a default value.

so two options, either have it pass in default value in `create_refs_and_properties` or just get rid of the param in `create_default_properties` with default value of true in the code directly

Reply:
The boolean value is needed as we need to pass in the value, this will become more clear in terms of migration, as we need to reconstruct the `CollectionProperty` when migrating over the `CollectionProperties`.  

I think it's best to keep default values for the `CollectionProperties`, so we don't let the user have to think about that at collection creation time.